server/archive.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+const lsArchive = require('ls-archive');
+
+function isArchive(path) {
+    return lsArchive.isPathSupported(path);
+}
+
+function listArchive(path) {
+    return new Promise((resolve, error) => {
+        try {
+            lsArchive.list(path, (err, entries) => {
+                if (err) {
+                    error(err);
+                    return;
+                }
+                const data = entries
+                    .map((z) => {
+                        return z.getPath();
+                    })
```

Comment:
Let's move out this function into `getPath` on top of the file (after require list):

```js
const getPath = (a) => a.getPath();
```

And use it with:

```js
const result = entries
    .map(getPath)
    .join('\n');

return result;
```